### DISCUSSION TITLE
dotnet publish to a win-* runtime under a Linux machine produces a different .exe than if published on windows

### DISCUSSION THREAD

**Discussion by sn4k3** [dotnet/runtime#66906] [Q&A]
I'm trying to change my windows `.ps` compilation script to a WSL `.sh` script which compile multiple runtimes.
When publishing to `win-x64` it produces a strange .exe, which will open a terminal and then execute the main app. Also the .exe lose it icon and remain with the default .exe icon.

My publish command: `dotnet publish $buildProject -o "$publishRuntimeDir" -c $buildWith -r $runtime -p:PublishReadyToRun=true --self-contained`

Is this a limitation on Linux and windows release must be strict published under windows?

**Comment by jkoritzinsky** [comment]
cc: @agocke around Win32 resources with the Windows apphost.

**Comment by vitek-karas** [comment]
This is a long standing issue: https://github.com/dotnet/runtime/issues/3828
For console apps publishing on any platform mostly works (if your app doesn't care about native resources, like product version and such), for GUI apps this doesn't work and you'll get the behavior you describe above.
